### PR TITLE
update ramda typings to sync them with types/npm-ramda

### PR DESCRIPTION
**Pull request by KiaraGrouwstra**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request). -- Some inference is still broken due to TS issues, see e.g. [here](https://github.com/Microsoft/TypeScript/issues/9949#issuecomment-292768693).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes). -- technically speaking currently there are a few cases of the [disguised type assertion](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/12321) where inference is not possible. These nonetheless however add value in the context of partial application, where casts would otherwise become significantly more verbose.
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/types/npm-ramda/
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dslint/dt.json" }`.

### CODE REVIEW

**Comment by tycho01** [comment]
Note this should make [PR 15668](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/15668) redundant (only updates a small bit).

On another note, Edwin (donnut) hasn't been active for a while... though I haven't touched the author list so far.

**Comment by tycho01** [comment]
Travis:
```
Error: /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/ramda/package.json should not specify name
    at checkPackageJson (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/lib/definition-parser.ts:144:10)
```
This appears to be a conflicting recommendation (demand?) between [DT](https://github.com/Microsoft/types-publisher/blob/master/src/lib/definition-parser.ts#L141-L146) and the [`package.json`](https://github.com/types/npm-quickly-copy-file/blob/master/package.json#L2) example recommended in the [@types guidelines](https://github.com/types/guidelines). @blakeembrey, is this a known thing? If so, are there other known incongruencies as well? Does @types rely on this `name` info which DT apparently wants out?

**Comment by blakeembrey** [comment]
@tycho01 Unfortunately, yes. I've tried to be vocal about the restrictions, but ultimately if we ever want to publish to DefinitelyTyped it looks like we have to restructure everything. Will probably end up being a script and publish to DefinitelyTyped since it's a non-trivial amount of work and the "DefinitelyTyped way" is mostly arbitrary and legacy, but I don't know if anyone is going to work on it (I'm not). The `package.json` in @types must have a name and preferably a version for NPM to use when installing (otherwise you can't install from git).

I think there's other restrictions with DefinitelyTyped too. You'll need to move all tests to `<library>-tests.ts`. You're also required to use `index.d.ts` too, can't specify the `typings` path in `package.json`. You also can't provide the contributors, homepage or other metadata in `package.json`, it must all be in the `index.d.ts` file header. I'm sure there's a couple more, but they're the main ones off the top of my head.

**Comment by tycho01** [comment]
@blakeembrey thanks for commenting. definitely a bit unfortunate.
I just ran into the `ramda has unused file ...` error too. ouch. I'm sure it's gonna have interesting opinions on how many of my files would be 'unused'. better preemptively delete the bunch of them on here.
I guess this does suck a bit, but it feels worth trying to go through manually for me for a bit now to get a sense of what it'd take.

**Comment by mhegazy** [comment]
@tycho01 can you please fix the failing CI tests.

File: types/ramda/package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (andy-ms) [inline types/ramda/package.json]: `package.json` isn't meant for this purpose. You shouldn't need a `package.json` at all.

Comment (tycho01) [inline types/ramda/package.json]: Good to know, thanks!

**Comment by andy-ms** [comment]
How is the status on ramda/ramda#1617 ?

**Comment by tycho01** [comment]
@andy-ms: sorry for not having updated there; I did now. Typings are at [types/npm-ramda](https://github.com/types/npm-ramda).

**Comment by mhegazy** [comment]
@tycho01 are you still interested in pursuing this change? if so we need the CI tests to pass before we merge it.

**Comment by tycho01** [comment]
@mhegazy: from #16072 I've taken it the main issue is in the process of getting a fix now:
> We currently infer `(x: {}) => { p: {}[] }`, but ideally we'd infer `<A>(x: A) => { p: A[] }`. That's next on the list.

When that lands presumably it should get a lot easier.

**Comment by mhegazy** [comment]
For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Comment by tycho01** [comment]
Fair enough.

**Pull request closed without merging**